quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/QuarkusPropertiesDistTest.java

```diff
@@ -153,14 +155,38 @@ void testInvalidSmallRyeKeyStorePathProperty(LaunchResult result) {
     @Test
     @Launch({ "start", "--http-enabled=true", "--hostname-strict=false",
             "--config-keystore=../../../../src/test/resources/keystore", "--config-keystore-password=secret" })
-    @Order(13)
+    @Order(12)
     void testSmallRyeKeyStoreConfigSource(LaunchResult result) {
         // keytool -importpass -alias kc.log-level -keystore keystore -storepass secret -storetype PKCS12 -v (with "debug" as the stored password)
         CLIResult cliResult = (CLIResult) result;
         assertTrue(cliResult.getOutput().contains("DEBUG"));
         cliResult.assertBuild();
     }
 
+    @Test
+    @BeforeStartDistribution(ForceRebuild.class)
```

Comment:
@Pepo48 does this mean that a build from one test is capturing a non-build time property, which then gets used by default in another test if that property isn't set explicitly?

cc @pedroigor

Reply:
@shawkins correct. I also remember that you a week or two ago mentioned that running multiple tests at once caused an issue while separately they passed. The same thing happened to here and it can be easily reproduced if you remove the @BeforeStartDistribution annotation.

I assume that this happens because of how the server configuration is currently persisted among individual test executions (afaict, there is no manual rebuild nor a properties cleanup in this specific case).

Reply:
@Pepo48 yes, I had attributed that to some error with config change I was trying to make.  

Just to make sure, let's use config-keystore, a runtime option, as an example for this behavior.  If it's set as an env variable, I do see that it is incidentially validated if you run kc.sh build - because Picocli.hasConfigChanges triggers the resolution of all properties.  That seems like a bug.  

If the build is successful, then I unset KC_CONFIG_KEYSTORE, it looks like both start and start --optimized will fail - hopefully this should mean we aren't leaking over the previous setting as a default.